### PR TITLE
Adding DegreeAPI endpoints

### PR DESCRIPTION
**Pull request by Feverfew**
## What does this PR do?
Implements #1033 
Adds department coursed endpoint.
Adds course modules endpoint, with instance and available/compulsory filtering.

## ✅ Pull Request checklist

- [ ] Is this code complete?
- [ ] Are tests done/modified?
- [ ] Are docs written for this PR? (if applicable)

## 🚨 Is this a breaking change for API clients?
No

## :squirrel: Deploy notes
Need to run migrations as part of deployment
Need the cache to update (probably?)
Need to rebuild the frontend (maybe?)
## Anything else
Don't let me merge this one.

### CODE REVIEW

**Comment by zipy124** [comment]
@ChristopherHammond13 This seems fine to merge now i think?

File: backend/uclapi/timetable/app_helpers.py [hunk 1]
```diff
@@ -516,6 +604,124 @@ def get_departmental_modules(department_id):
...
+    :type dept_modules: dict
+    :param course_id: course id
+    :type course_id: str
+    :param query_params: dictionary of validated query parameters
+    :type query_params: django.http.QueryDict
+
+    """
+    modules = get_cache("crscompmodules")
+    for compulsory in modules.objects \
+                             .filter(courseid=course_id, setid=_SETID) \
```
Comment (ChristopherHammond13) [inline backend/uclapi/timetable/app_helpers.py]: This is such beautiful code 😻

File: backend/uclapi/timetable/app_helpers.py [hunk 2]
```diff
@@ -13,12 +14,21 @@
...
-                     StudentsB, StumodulesA, StumodulesB,  TimetableA,
-                     TimetableB, WeekmapnumericA, WeekmapnumericB,
-                     WeekstructureA, WeekstructureB,
-                     CminstancesA, CminstancesB)
+from .models import (
+    CminstancesA, CminstancesB,
+    CourseA, CourseB,
+    CrscompmodulesA, CrscompmodulesB,
+    CrsavailmodulesA, CrsavailmodulesB
+    DeptsA, DeptsB,
```
[the remark this answers is not in the document]
Comment (ChristopherHammond13) [reply backend/uclapi/timetable/app_helpers.py]: This has been fixed now

**Comment by ChristopherHammond13** [comment]
Merging this at last!! 🎉

**Pull request merged**